File: verify_home.js
import { chromium } from 'playwright';

(async () => {
	const browser = await chromium.launch();
	const page = await browser.newPage();

	try {
		// Wait for server to start (simple delay for this example, or poll)
		await page.waitForTimeout(5000);

		await page.goto('http://localhost:5173/');

		// Verify H1
		const heading = page.getByRole('heading', { level: 1 });
		await heading.waitFor({ state: 'visible' });

		// Verify it says "GraphQL Endpoint Explorer"
		const text = await heading.textContent();
		if (text?.trim() !== 'GraphQL Endpoint Explorer') {
			throw new Error(`Expected heading to be 'GraphQL Endpoint Explorer', got '${text}'`);
		}

		await page.screenshot({ path: '/home/jules/verification/verification.png' });
		console.log('Verification successful, screenshot saved.');
	} catch (e) {
		console.error('Verification failed:', e);
		process.exit(1);
	} finally {
		await browser.close();
	}
})();

File: verification/verify_home.py
from playwright.sync_api import sync_playwright

def verify_endpoints():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        try:
            print("Navigating to home page...")
            page.goto("http://localhost:5173")
            page.wait_for_load_state("networkidle")

            # Click Get Started
            print("Clicking Get Started...")
            page.get_by_role("link", name="Get Started").click()
            page.wait_for_url("**/endpoints")
            page.wait_for_load_state("networkidle")

            print("Taking screenshot of endpoints...")
            page.screenshot(path="verification/endpoints.png")
            print("Screenshot saved to verification/endpoints.png")

        except Exception as e:
            print(f"Error: {e}")
        finally:
            browser.close()

if __name__ == "__main__":
    verify_endpoints()
